src/mlpack/methods/reinforcement_learning/environment/continuous_double_pole_cart.hpp

```diff
@@ -0,0 +1,374 @@
+    const arma::colvec& Encode() const { return data; }
+
+    //! Dimension of the encoded state.
+    static constexpr size_t dimension = 6;
+
+   private:
+    //! Locally-stored state data.
+    arma::colvec data;
+  };
+
+  /**
+   * Implementation of action of Continuous Double Pole Cart.
+   */
+  struct Action
+  {
+    double action = 0.0;
+    // Storing degree of freedom
+    const int size = 1;
+  };
+
+  /**
+   * Construct a Double Pole Cart instance using the given constants.
+   *
+   * @param m1 The mass of the first pole.
+   * @param m2 The mass of the second pole.
+   * @param l1 The length of the first pole.
+   * @param l2 The length of the second pole.
+   * @param gravity The gravity constant.
+   * @param massCart The mass of the cart.
+   * @param length The length of the pole.
+   * @param forceMag The magnitude of the applied force.
+   * @param tau The time interval.
+   * @param thetaThresholdRadians The maximum angle.
+   * @param xThreshold The maximum position.
+   * @param doneReward Reward recieved by agent on success.
+   * @param maxSteps The number of steps after which the episode
+   *    terminates. If the value is 0, there is no limit.
+   */
+  ContinuousDoublePoleCart(const double m1 = 0.1,
+                             const double m2 = 0.01,
+                             const double l1 = 0.5,
+                             const double l2 = 0.05,
+                             const double gravity = 9.8,
+                             const double massCart = 1.0,
+                             const double forceMag = 10.0,
+                             const double tau = 0.02,
+                             const double thetaThresholdRadians = 36 * 2 *
+                                3.1416 / 360,
+                             const double xThreshold = 2.4,
+                             const double doneReward = 0.0,
+                             const size_t maxSteps = 0) :
+      m1(m1),
+      m2(m2),
+      l1(l1),
+      l2(l2),
+      gravity(gravity),
+      massCart(massCart),
+      forceMag(forceMag),
+      tau(tau),
+      thetaThresholdRadians(thetaThresholdRadians),
+      xThreshold(xThreshold),
+      doneReward(doneReward),
+      maxSteps(maxSteps),
+      stepsPerformed(0)
+  { /* Nothing to do here */ }
+
+  /**
+   * Dynamics of Continuous Double Pole Cart instance. Get reward and next
+   * state based on current state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @return reward, it's always 1.0.
+   */
+  double Sample(const State& state,
+                const Action& action,
+                State& nextState)
+  {
+    // Update the number of steps performed.
+    stepsPerformed++;
+
+    arma::vec dydx(6, arma::fill::zeros);
+    for (size_t i = 0; i < 2; i++)
+    {
+      dydx[0] = state.Velocity();
+      dydx[2] = state.AngularVelocity(1);
```

Comment:
Looks like the ``state.AngularVelocity`` and ``state.AngularVelocity`` is missing the ``I``parameter.